src/components/App.js

```diff
@@ -1,14 +1,66 @@
-import ButtonPanel from './ButtonPanel';
+import React, { Component } from 'react';
 import Display from './Display';
+import ButtonPanel from './ButtonPanel';
 import calculate from '../logic/calculate';
 
-calculate(4, 5, '+');
+class App extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+      lastClicked: null,
+    };
+  }
+
+  handleClick = (buttonName) => {
+    const {
+      total, next, operation, lastClicked,
+    } = this.state;
+    const operations = ['+', '-', '÷', 'X', '='];
+    if (buttonName === '.') {
+      this.setState({
+        next: next == null ? `0${buttonName}` : next + buttonName,
+      });
+    } else if (buttonName in ['1', '2', '3', '4', '5', '6', '7', '8', '9', '0']) {
+      this.setState({
+        next: next == null ? buttonName : next + buttonName,
+      });
+    } else {
+      switch (buttonName) {
+        case '+':
+        case '-':
+        case '÷':
+        case 'X':
+        case '=':
+          if (total === null || total === 'undefined') {
+            this.setState({
+              total: next,
+              next: null,
+              operation: buttonName,
+            });
+          } else if (!operations.includes(lastClicked)) {
+            this.setState(calculate({ total, next, operation }, buttonName));
+          }
+          break;
+        default:
+          this.setState(calculate({ total, next, operation }, buttonName));
```

Comment:
let's improve this part of the code.

If you can take a step backward, and check the instructions from the pathwright, you will see that all you need to do in this file is to create the `Constructor` and bind the `handleClick` method.
The if-else statement and switch cases should be inside `Calculate.js`

I don't think you need `lastClicked`

Example below;
```
class App extends Component {  
     constructor() {    
        super();    
        this.state = {     
            total: null, ......
        };
       this.handleClick = this.handleClick.bind(this);
    }
    
    handleClick(buttonName) {
    ......
    }
  .....
}
```
![image](https://user-images.githubusercontent.com/31897434/118756317-8b66ad80-b862-11eb-9bdd-d3f2446b7e04.png)